### PR TITLE
Switch to GH Workflows for CI

### PR DESCRIPTION
**Pull request by ramereth**
This switches this cookbook to use GH Workflows instead of TravisCI to match what Sous Chef uses. This is also intended to provide a few other minor cleanups.

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by ramereth** [comment]
@tas50 this should be ready for you to review!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@master
+      - name: Run yaml Lint
+        uses: actionshub/yamllint@master
+
+  mdl:
```
Comment (tas50) [inline .github/workflows/ci.yml:29]: Skip this.

File: README.md [hunk 2]
```diff
@@ -277,17 +278,19 @@ And then change the action to "remove":
...
 
 - You want to search a different data bag specific to a role such as mail. You may change the `data_bag` searched.
+
   ```ruby
   data_bag `mail`
   ```
 
-- You want to search for a different group attribute named `postmaster`. You may change the `search_group` attribute. This
-attribute defaults to the LWRP resource name.  
+- You want to search for a different group attribute named `postmaster`. You may change the `search_group` attribute. This attribute defaults to the LWRP resource name.
```
Comment (tas50) [inline README.md:286]:
```suggestion
- You want to search for a different group attribute named `postmaster`. You may change the `search_group` attribute. This attribute defaults to the resource name.
```

File: README.md [hunk 3]
```diff
@@ -277,17 +278,19 @@ And then change the action to "remove":
...
+
   ```ruby
   search_group `postmaster`
   ```
 
 - You want to add the users to a security group other than the lightweight resource name. You may change the `group_name` attribute. This attribute also defaults to the LWRP resource name.
```
Comment (tas50) [inline README.md:292]:
```suggestion
- You want to add the users to a security group other than the lightweight resource name. You may change the `group_name` attribute. This attribute also defaults to the resource name.
```

File: kitchen.yml [hunk 4]
```diff
@@ -24,16 +27,12 @@ platforms:
   - name: opensuse-leap-15
   - name: macosx-10.13
     driver:
-      box: chef/macosx-10.13 # private box in Chef's Atlas account
+      box: chef/macosx-10.13  # private box in Chef's Atlas account
       provider: vmware_fusion
   - name: opensuse-leap-15
   - name: ubuntu-16.04
-  - name: ubuntu-16.04-chef-12.7
```
Comment (tas50) [inline kitchen.yml:31]: You probably want to leave one here that's 12.15 so the oldest supported infra client can be tested

**Pull request merged**